plugin/pkg/scheduler/schedulercache/cache.go

```diff
@@ -74,14 +74,23 @@ func newSchedulerCache(ttl, period time.Duration, stop <-chan struct{}) *schedul
 	}
 }
 
-func (cache *schedulerCache) GetNodeNameToInfoMap() (map[string]*NodeInfo, error) {
-	nodeNameToInfo := make(map[string]*NodeInfo)
+func (cache *schedulerCache) UpdateNodeNameToInfoMap(nodeNameToInfo map[string]*NodeInfo) error {
```

Comment:
Another idea I have was to keep the nodeInfoMap immutable. so that we can access the info map without locking/copying or versioning. but it might be overkill.